internal/dag/builder_test.go

```diff
@@ -10993,6 +10993,169 @@ func TestBuilderRunsProcessorsInOrder(t *testing.T) {
 	assert.Equal(t, []string{"foo", "bar", "baz", "abc", "def"}, got)
 }
 
+func TestDefaultHeadersPolicies(t *testing.T) {
```

Comment:
@youngnick Felt the need to add more testing here. The file is quite large, but there weren't really any good tests for the global headers policies (that I could find).

It's a bit repetitive, but I needed the testing to validate the changes I was making

Reply:
These tests look great.

Reply:
I know the rest of the tests in this file use the fixture-indexing standard, but I'd encourage swapping these to more descriptive names, like `ingressKuardDefault` or something? `proxyMultipleBackends` is a good example.